kunlar/13-kun-Net-Programma/13-net-programma.md

```diff
@@ -0,0 +1,251 @@
+# Networked programs
+
+Shu paytgacha biz programma tuzganimizda. Ular malumotni kompyuter o'zidan o'qidi/oldi. Yoki fayllar bilan 
+ishlaganimizda fayllar kompyuterimizda turgan edi. Lekin, internetga esa malumotlar ko'p manba'lardan kelishi 
+mumkin. 
+
+Bugin, biz o'zimiz web brovser (google chrome, opera, safari) rolini o'ynaymiz. Va veb sahifalarni HTTP 
+(Hypertext Transfer Protocol) orqali olamiz. Keyin ulardan malumotlarni o'qib, ulardan keraklisini ajratib olamiz (parse).
+
+
+
+## HTTP Protokol
+Veb sahifani ishlatadigan netvork protokoli aslida juda oson. Pythonda `socket` degan biblioteka bor. Bu biblioteka
+netvork aloqalarni taminlashni va soketlardan maumotlarni olishni juda osonlashtiradi.
+
+Soket bu xuddi faylga o'xshaydi. Farqli tarafi shundaki, bitta soket ikki programma o'rtasida ikki tomonlama aloqani
+taminlaydi. Agar, soketga biron malumot yozilsa, u programmaga yuboriladi. Agar malumotni soketdan o'qisangiz
+demak u malumotni boshqa programma yuborganini bildiradi.  Quyidagicha tasvirlash mumkin:
+
+<p align="center">
+    <img src="./imgs/soket.png" height=200>
+</p>
+
+
+Aytaylik soketga hech qanday programma malumot yubormagan, unda sizni programmangiz baribir malumot kelishini shunchaki kutadi.
+Aytaylik 2 ta programma soketga hech qanday malumot yubormay uzoq vaqt malumot yubormasa, ular bir birini juda uzoq kutib qoladi. 
+Shuning uchun internat orqali aloqa qiladigan programmalar uchun protokol joriy qilingan.
+
+Protokol bu aniq qoidalar to'plamiki bunda qaysi programma birinchi malumot yuboradi, nima qiladi,va keyin malumotga
+qanaqa javob bo'lish kerak, keyingi malumotni kim yuboradi va hokazo. Shundayki, 2 ta programma bir biri bilan
+skoet orqali navbatma va navbat aloqa qilishdir. 
+
+Hozrida juda ko'p protokollar mavjud. Shulardan HTTP quyidagi linkda berilgan 
```

Comment:
Shulardan HTTP quyidagi linkda berilgan - Shulardan biri HTTP, quyidagi linkda berilgan.
deb yozsak yaxshiroq

Reply:
To'g'irlandi. Rahmat.